src/Sylius/Component/Core/Promotion/Checker/NthOrderRuleChecker.php

```diff
@@ -48,6 +48,10 @@ public function isEligible(PromotionSubjectInterface $subject, array $configurat
             throw new UnsupportedTypeException($subject, OrderInterface::class);
         }
 
+        if (!isset($configuration['nth']) || !is_int($configuration['nth'])) {
+            return false;
+        }
+
         if (null === $customer = $subject->getCustomer()) {
```

Comment:
Not a part of this PR, but IMO putting an assignment inside a condition is hardly readable.

```php
$customer = $subject->getCustomer();
if (null === $customer) {
  return false;
}
```